File: web/src/api/requestApi.ts
import { axiosClient } from './axiosClient';
import type { MaterialRequest, CreateRequestDto, FulfillRequestDto } from '../types';

export const requestApi = {
  getById: async (id: string): Promise<MaterialRequest> => {
    const response = await axiosClient.get(`/requests/${id}`);
    return response.data.data;
  },

  create: async (data: CreateRequestDto): Promise<MaterialRequest> => {
    const response = await axiosClient.post('/requests', data);
    return response.data.data;
  },

  fulfill: async (id: string, data: FulfillRequestDto): Promise<MaterialRequest> => {
    const response = await axiosClient.post(`/requests/${id}/fulfill`, data);
    return response.data.data;
  },

  updateMaterial: async (id: string, data: FulfillRequestDto): Promise<MaterialRequest> => {
    const response = await axiosClient.put(`/requests/${id}/material`, data);
    return response.data.data;
  },

  delete: async (id: string): Promise<void> => {
    await axiosClient.delete(`/requests/${id}`);
  },
};

export default requestApi;

File: mobile/ResearchCenter/app/src/main/java/com/example/researchcenter/shared/api/RequestApi.kt
package com.example.researchcenter.shared.api

import com.example.researchcenter.shared.model.ApiResponse
import com.example.researchcenter.shared.model.MaterialRequest
import retrofit2.Call
import retrofit2.http.*

interface RequestApi {
    @POST("repositories/{repoId}/requests")
    fun createRequest(@Path("repoId") repoId: Long, @Body body: Map<String, String>): Call<ApiResponse<MaterialRequest>>

    @POST("requests/{id}/fulfill")
    fun fulfillRequest(@Path("id") id: Long, @Body body: Map<String, Long>): Call<ApiResponse<MaterialRequest>>

    @PUT("requests/{id}/material")
    fun updateFulfillment(@Path("id") id: Long, @Body body: Map<String, Long>): Call<ApiResponse<MaterialRequest>>

    @DELETE("requests/{id}")
    fun deleteRequest(@Path("id") id: Long): Call<ApiResponse<Any>>
}
